backup_restorer.cc

```diff
@@ -315,4 +315,92 @@ void IndexedRestorer::saveData( int64_t offset, void * data, size_t size ) const
   }
 }
 
+void IndexedRestorer::restore( int64_t offset, DataSink * output, size_t size ) const
```

Comment:
This is almost exact copy of `saveData` method above.
Lets implement one using another, e.g. implement `saveData` using `restore` with custom DataSink.